src/Sulu/Bundle/TrashBundle/Domain/Factory/TrashItemFactory.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TrashBundle\Domain\Factory;
+
+use Sulu\Bundle\TrashBundle\Domain\Model\TrashItemInterface;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Symfony\Component\Security\Core\Security;
+
+class TrashItemFactory implements TrashItemFactoryInterface
```

Comment:
Would mark this service as final as we will not support extending here as official extension point:

```suggestion
final class TrashItemFactory implements TrashItemFactoryInterface
```

(if we keep this as own service and don't add the create method on the repository)